bots/elasticsearch/bot.py

```diff
@@ -193,7 +213,10 @@ def run(self, chunk_size=500):
             else:
                 most_recent_result = pendulum.parse(self.get_most_recently_modified())
                 logger.info('Looking for %ss that have been modified after %s', model, most_recent_result)
-                qs = model.objects.filter(date_modified__gt=most_recent_result).values_list('id', flat=True)
+                q = Q(date_modified__gt=most_recent_result)
+                if hasattr(model, 'subjects') and hasattr(model, 'subject_relations'):
+                    q = q | Q(subjects__date_modified__gt=most_recent_result) | Q(subject_relations__date_modified__gt=most_recent_result)
```

Comment:
```sql
EXPLAIN SELECT share_creativework.type, share_creativework.id, share_creativework.title, share_creativework.description, share_creativework.is_deleted, share_creativework.date_published, share_creativework.date_updated, share_creativework.free_to_read_type, share_creativework.free_to_read_date, share_creativework.rights, share_creativework.language, share_creativework.date_created, share_creativework.change_id, share_creativework.date_modified, share_creativework.version_id, share_creativework.same_as_id, share_creativework.same_as_version_id, share_creativework.extra_id, share_creativework.extra_version_id
FROM share_creativework
  LEFT OUTER JOIN share_throughsubjects ON (share_creativework.id = share_throughsubjects.creative_work_id)
  LEFT OUTER JOIN share_subject ON (share_throughsubjects.subject_id = share_subject.id)
WHERE (share_creativework.type IN ('share.software', 'share.project', 'share.patent', 'share.poster', 'share.conferencepaper', 'share.preprint', 'share.creativework', 'share.article', 'share.report', 'share.workingpaper', 'share.repository', 'share.retraction', 'share.publication', 'share.dissertation', 'share.book', 'share.registration', 'share.presentation', 'share.dataset', 'share.thesis')
       AND (
         share_creativework.date_modified > '2017-06-08T18:39:19.629395+00:00'
         OR share_subject.date_modified > '2017-06-08T18:39:19.629395+00:00'
         OR share_throughsubjects.date_modified > '2017-06-08T18:39:19.629395+00:00'
       )
)
```
(ugh typedmodels)
```
Hash Left Join  (cost=47.95..3994.97 rows=1 width=525)
  Hash Cond: (share_throughsubjects.subject_id = share_subject.id)
  Filter: ((share_creativework.date_modified > '2017-06-08 14:39:19.629395-04'::timestamp with time zone) OR (share_subject.date_modified > '2017-06-08 14:39:19.629395-04'::timestamp with time zone) OR (share_throughsubjects.date_modified > '2017-06-08 14:39:19.629395-04'::timestamp with time zone))
  ->  Merge Left Join  (cost=0.57..3495.88 rows=21257 width=537)
        Merge Cond: (share_creativework.id = share_throughsubjects.creative_work_id)
        ->  Index Scan using share_creativework_pkey on share_creativework  (cost=0.29..3082.48 rows=21257 width=525)
              Filter: ((type)::text = ANY ('{share.software,share.project,share.patent,share.poster,share.conferencepaper,share.preprint,share.creativework,share.article,share.report,share.workingpaper,share.repository,share.retraction,share.publication,share.dissertation,share.book,share.registration,share.presentation,share.dataset,share.thesis}'::text[]))
        ->  Index Scan using share_throughsubjects_creative_work_id_9bef7ff0 on share_throughsubjects  (cost=0.28..288.39 rows=6872 width=16)
  ->  Hash  (cost=32.17..32.17 rows=1217 width=12)
        ->  Seq Scan on share_subject  (cost=0.00..32.17 rows=1217 width=12)
```
Seems pretty decent? There is an index on `share_subject.date_modified`, for the record, but I guess postgres didn't feel like using it.